### PR TITLE
Add discourse docs module

### LINKED ISSUE #9: Add the discourse-docs module

### PR DESCRIPTION
**Pull request by anthonydillon**
## Done
Add discourse docs module to load content from the [discourse docs category[(https://discourse.dqlite.io/t/dqlite-documentation/21).

## QA
- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: http://0.0.0.0:8037/docs
- Check it renders correctly
- Click on the pages and check they work too

## Issue / Card

Fixes #9

### CODE REVIEW

File: webapp/app.py [hunk 1]
```diff
@@ -12,22 +18,37 @@
     template_500="500.html",
 )
 
+doc_parser = DocParser(
+    api=DiscourseAPI(base_url="https://discourse.dqlite.io/"),
+    index_topic_id=21,
+    category_id=2,
```
Comment (tbille) [inline webapp/app.py:24]:
```suggestion
    category_id=5,
```

Comment (anthonydillon) [reply webapp/app.py]: Done

File: static/sass/_layout_docs.scss [hunk 2]
```diff
@@ -0,0 +1,82 @@
+@mixin maas-l-docs {
```
[the remark this answers is not in the document]
Comment (anthonydillon) [reply static/sass/_layout_docs.scss]: I'll add try out the new layout it was easier to start from here.

[the remark this answers is not in the document]
Comment (anthonydillon) [reply static/sass/_layout_docs.scss]: There you go

File: templates/docs/document.html [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  /**
+  Toggles the expanded/collapsed classed on side navigation element.
+
+  @param {HTMLElement} sideNavigation The side navigation element.
+  @param {Boolean} show Whether to show or hide the drawer.
+*/
+function toggleDrawer(sideNavigation, show) {
+  if (sideNavigation) {
+    if (show) {
+      sideNavigation.classList.remove('is-collapsed');
```
Comment (hatched) [inline templates/docs/document.html:52]: Just FYI `classList` has a `toggle` method that you could use to simplify this function.

Comment (solazio) [reply templates/docs/document.html:52]: @bartaz I guess this should be done in vanilla as well.

Comment (bartaz) [reply templates/docs/document.html:52]: If I  remember correctly `toggle` had some support issues in IE. It didn't support the second bool param that say if toggle should be on/off.

**Comment by solazio** [comment]
The table of contents should be removed as the links don't work. https://discourse.dqlite.io/t/documentation-faq/22

File: templates/docs/document.html (hunk 3 above)
Comment (anthonydillon) [reply templates/docs/document.html:52]: I suppose that would work and should'nt get out of sync?

Comment (anthonydillon) [reply templates/docs/document.html:52]: I guess I am wondering what the trade-off's of being prescriptive over less code

File: webapp/app.py [hunk 4]
```diff
@@ -12,22 +18,37 @@
...
 
 @app.route("/")
 def index():
     return render_template("index.html")
 
 
-@app.route("/docs")
-def docs():
-    return render_template("docs/index.html")
+# @app.route("/docs")
```
Comment (tbille) [inline webapp/app.py:42]: remove comments

Comment (anthonydillon) [reply webapp/app.py:42]: Done

**Comment by anthonydillon** [comment]
@solazio Good spot, TOC is gone

**Comment by bartaz** [comment]
@solazio That's the plan, we have it in backlog already for next iteration https://github.com/canonical-web-and-design/vanilla-framework/issues/3021

File: templates/docs/document.html (hunk 3 above)
Comment (bartaz) [reply templates/docs/document.html:52]: Sometimes (like on window resize) we want to just remove those classes (hide the drawer and overlay), not toggle them.
We could have a separate function to do this. But that, again, is more code to do kinda the same thing.

After rewriting all Vanilla JS examples into longer, more prescriptive, IE consistent versions I'm kinda still in this mindset ;)
I'm all for writing less, but I think it's really safe only when we babel.

**Pull request merged**